block/manager.go

```diff
@@ -164,6 +167,16 @@ func NewManager(
 		conf.BlockTime = defaultBlockTime
 	}
 
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	maxBlobSize, err := dalc.DA.MaxBlobSize(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if genesis.ConsensusParams.Block.MaxBytes >= int64(maxBlobSize) {
+		genesis.ConsensusParams.Block.MaxBytes = int64(maxBlobSize) - blockProtocolOverhead
+	}
+
```

Comment:
Let's also pass it through `ConsensusParamsValidateBasic` and make sure there's no errors.